### PR TITLE
feat(rule): add `maxRetryTime` option

### PR DESCRIPTION
**Pull request by chick-p**
Fix https://github.com/textlint-rule/textlint-rule-no-dead-link/issues/135

I've not added this test case. 
Because I expect this test case occurs timeout.
Ref. https://github.com/textlint/textlint/issues/781

### CODE REVIEW

File: src/no-dead-link.js [hunk 1]
```diff
@@ -186,9 +187,14 @@ const createCheckAliveURL = (ruleOptions) => {
 
       // try to fetch again if not reach max retry count
       if (currentRetryCount < maxRetryCount) {
-        // exponential retry
-        // 0ms -> 100ms -> 200ms -> 400ms -> 800ms ...
-        await waitTimeMs((currentRetryCount ** 2) * 100);
+        const retrySeconds = res.headers.get('Retry-After');
```
Comment (azu) [inline src/no-dead-link.js:190]: I think that we need to prefer it if the response header includes `Retry-After`.

probably, next code.
```js
const retrySeconds = res.headers.get('Retry-After');
// If the response has `Retry-After` header, prefer it
// else exponential retry: 0ms -> 100ms -> 200ms -> 400ms -> 800ms ...
const retryWaitTimeMs = retrySeconds !== null ? retrySeconds * 1000 : currentRetryCount ** 2 * 100;
const maxRetryTimeMs = ruleOptions.maxRetryTime * 1000;
if (retryWaitTime <= maxRetryTimeMs) {
  await waitTimeMs(currentRetryCount ** 2 * 100);
}
return isAliveURI(uri, 'GET', maxRetryCount, currentRetryCount + 1);
```

File: src/no-dead-link.js [hunk 2]
```diff
@@ -20,7 +20,8 @@ const DEFAULT_OPTIONS = {
   interval: 500, // The length of time in milliseconds before the interval count resets. Must be finite. [Experimental]
   intervalCap: 8, // The max number of runs in the given interval of time. [Experimental]
   keepAlive: false, // {boolean} if it is true, use keepAlive for checking request [Experimental]
-  userAgent: 'textlint-rule-no-dead-link/1.0' // {String} a UserAgent
+  userAgent: 'textlint-rule-no-dead-link/1.0', // {String} a UserAgent,
+  maxRetryTime: 0, // (number) The max of waiting seconds for retry, if response returns `After-Retry` header.
```
Comment (azu) [inline src/no-dead-link.js:24]: This default value means that we will ignore `Retry-After` by default.
I think that we should consider `Retry-After` header if the server response it.

Increase maxRetryTime to 10 or 30?

File: src/no-dead-link.js (hunk 1 above)
Comment (chick-p) [reply src/no-dead-link.js:190]: @azu 
Thank you for your comment.
I rewrite the code as you said.

File: src/no-dead-link.js (hunk 2 above)
Comment (chick-p) [reply src/no-dead-link.js]: I've set it to `10`.

I've set `0` to not impact those who are already using it.
But I understand that `After-Retry` value should be used.

**Comment by azu** [comment]
Thanks!

**Comment by azu** [comment]
Release [Release v4.8.0 · textlint-rule/textlint-rule-no-dead-link](https://github.com/textlint-rule/textlint-rule-no-dead-link/releases/tag/v4.8.0) 🎉

**Pull request merged**